### ISSUE TITLE
Simplify the repository file structure and the versioning scheme (depends on #828)

### ISSUE THREAD

**Issue by hackaugusto** [raiden-network/raiden-contracts#524]
This repo is quite convoluted:

- solidity code and static data is inside the python package
- there is un-versioned static data 
- the versioned static data are using a scheme which will pollute the python's package root-dir

Proposal:

- move the smart contract code to the top-level
- create a folder named data in the top-level, inside this we will have the versioned smart contracts deployment data, possible schemes:

1. `./data/<network>/<smart_contract_version>/<smart_contract_name>.json`
2. `./data/<network>/<smart_contract_name>/<smart_contract_version>/deploy.json`
3. `./data/<network>/<bundle_version>/<smart_contract_name>-<smart_contract_version>.json`

All of the above assumes that deploys to different networks are independent, therefore the data is grouped per network. This is because eventually a test network will not be supported anymore.

Option 1 assumes that all smart contracts will be deployed in lock step. Which I think it is a bad idea, the `SecretRegistry` most likely should be shared across all future token networks.

Option 2 solves the above, but doesn't tell you which version of the smart contract works together

Option 3 is my preferred option. Smart contracts are deployed and tested in bundles.

I believe the smart contract code should follow the same pattern describe above, instead of starting at `./data` it should be `./smartcontracts`

For python package release, I'd recommend using `raiden-smartcontracts-<network>-<version>` with a single version of the smart contracts, or `raiden-smartcontracts` with all the smart contracts, I prefer option 2 since it's the simplest and the amount of binary data is probably negligible.

**Comment by pirapira** [comment]
> solidity code and static data is inside the python package

Which one do you need in the python package?

**Comment by pirapira** [comment]
About the deployment data, option 3 is my favorite too.

**Comment by hackaugusto** [comment]
> Which one do you need in the python package?

The solidity code is not necessary if we have the static data. The ABIs, addresses, and bytecode are there.

**Comment by pirapira** [comment]
I guess we should
* let each contract have its own version
* stop putting deployment data for different contracts in a single file
* allow deployment bundles to share a single deployment.  When I upgrade deployment bundles, I sometimes want to keep using the same `SecretRegistry` deployment at the same address, but start using a new `TokenNetworkRegistry` deployment at a new address.

**Comment by pirapira** [comment]
Take into account #818 .  Only one SecretRegistry should be deployed per chain.

**Comment by pirapira** [comment]
# [short title of solved problem and solution]

* Status: [proposed | rejected | accepted | deprecated | … | superseded by [ADR-0005](0005-example.md)] 
* Deciders: [list everyone involved in the decision] 
* Date: [YYYY-MM-DD when the decision was last updated] 

Technical Story: [description | ticket/issue URL] 

## Context and Problem Statement

[Describe the context and problem statement, e.g., in free form using two to three sentences. You may want to articulate the problem in form of a question.]

## Decision Drivers 

* [driver 1, e.g., a force, facing concern, …]
* [driver 2, e.g., a force, facing concern, …]
* … 

## Considered Options

* [option 1]
* [option 2]
* [option 3]
* … 

## Decision Outcome

Chosen option: "[option 1]", because [justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)].

### Positive Consequences 

* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
* …

### Negative Consequences 

* [e.g., compromising quality attribute, follow-up decisions required, …]
* …

## Pros and Cons of the Options 

### [option 1]

[example | description | pointer to more information | …] 

* Good, because [argument a]
* Good, because [argument b]
* Bad, because [argument c]
* … 

### [option 2]

[example | description | pointer to more information | …] 

* Good, because [argument a]
* Good, because [argument b]
* Bad, because [argument c]
* … 

### [option 3]

[example | description | pointer to more information | …] 

* Good, because [argument a]
* Good, because [argument b]
* Bad, because [argument c]
* … 

## Links 

* [Link type] [Link to ADR] 
* …

**Comment by pirapira** [comment]
Also consider https://github.com/raiden-network/raiden-contracts/issues/822 ,  maybe on test networks we want deployments with different settlement periods.

**Comment by pirapira** [comment]
I think I should first choose a version scheme https://github.com/raiden-network/raiden-contracts/issues/828 and then define the directory structure.

**Comment by pirapira** [comment]
* The version scheme will change according to #828 
* The layout in the file system will not change much.
    * I'll keep duplicating files into `data_x.y.z`.
    * I talked with @rakanalh and @hackaugusto at length, but the only motivations for changes are
        * desire to represent conceptional factorization as file system structure (like database normalization)
        * detest at duplication
    * The file system structure doesn't need to represent the conceptional factorization (just because we have sources and bundles, we don't need `sources/` and `bundles/`).  Nobody wants a list of all versions of TokenNetwork.
    * Copies are simple.
        * If you choose to have A1 and A2 as copies, you don't design anything any further
        * Once you choose to have a single file behind two paths A1 and A2, you need to choose to have (1) a link from A1 to A2 (2) from A2 to A1 (3) from both A1 and A2 to somewhere
    * I never need to edit a Solidity source that's used in the past.  I never need to modify TokenNetwork.sol at contracts_version 0.4.0.  That's why I never suffer from the duplication.
